pkg/noderesourcetopology/filter.go

```diff
@@ -0,0 +1,263 @@
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "NodeResourceTopologyMatch"
+)
+
+var _ framework.FilterPlugin = &TopologyMatch{}
+
+type nodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type PolicyHandler func(pod *v1.Pod, zoneMap topologyv1alpha1.ZoneList) *framework.Status
+
+type PolicyHandlerMap map[apiconfig.TopologyManagerPolicy]PolicyHandler
+
+// TopologyMatch plugin which run simplified version of TopologyManager's admit handler
+type TopologyMatch struct {
+	policyHandlers PolicyHandlerMap
+	lister         listerv1alpha1.NodeResourceTopologyLister
+	namespace      string
+}
+
+type NUMANode struct {
+	NUMAID    int
+	Resources v1.ResourceList
+}
+
+type NUMANodeList []NUMANode
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (tm *TopologyMatch) Name() string {
+	return Name
+}
+
+// getTopologyPolicies return true if we're working with such policy
+func getTopologyPolicies(nodeTopology *topologyv1alpha1.NodeResourceTopology, nodeName string) []apiconfig.TopologyManagerPolicy {
+	policies := make([]apiconfig.TopologyManagerPolicy, 0)
+	for _, policy := range nodeTopology.TopologyPolicies {
+		policies = append(policies, apiconfig.TopologyManagerPolicy(policy))
+	}
+	return policies
+}
+
+func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
+	res := make(v1.ResourceList)
+	for _, resInfo := range zone.Resources {
+		quantity, err := resource.ParseQuantity(resInfo.Allocatable.String())
+		if err != nil {
+			klog.Errorf("Failed to parse %s", resInfo.Allocatable.String())
+			continue
+		}
+		res[v1.ResourceName(resInfo.Name)] = quantity
+	}
+	return res
+}
+
+func SingleNUMAContainerLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneList) *framework.Status {
+	klog.V(5).Infof("Single NUMA node handler")
+	containers := []v1.Container{}
+	containers = append(pod.Spec.InitContainers, pod.Spec.Containers...)
+
+	// prepare NUMANodes list from zoneMap
+	nodes := createNUMANodeList(zones)
+	qos := v1qos.GetPodQOS(pod)
+	for _, container := range containers {
+		bitmask := bm.NewEmptyBitMask()
+		bitmask.Fill()
+
+		checkResourcesForNUMANodes(bitmask, nodes, container.Resources.Requests, qos)
+		if bitmask.IsEmpty() {
+			// definitly we can't align container, so we can't align a pod
+			return framework.NewStatus(framework.Unschedulable, fmt.Sprintf("Cannot align container: %s", container.Name))
+		}
+	}
+	return nil
+}
+
+func checkResourcesForNUMANodes(bitmask bm.BitMask, nodes NUMANodeList, resources v1.ResourceList, qos v1.PodQOSClass) {
```

Comment:
I may not be the right person validating the logic here, @swatisehgal could you help review the correctness and ensure it's tested properly? Thanks.

Reply:
It may be more readable to return a `bm.BitMask`, and remove `bitmask bm.BitMask` in the input args, and also we don't need to init the bitmask on the caller side.

Reply:
yes, not so golangish, to be a more natural for golang it's better to return instead of using *out* argument.